crates/superposition_types/src/lib.rs

```diff
@@ -243,9 +244,59 @@ impl Display for RegexEnum {
     }
 }
 
-#[derive(Clone, Deserialize)]
+#[derive(Clone, Debug, Deserialize)]
+pub enum HeadersEnum {
+    ConfigVersion,
+}
+
+impl fmt::Display for HeadersEnum {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            HeadersEnum::ConfigVersion => write!(f, "Config-Version"),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub enum HttpMethod {
+    Get,
+    Post,
+    Put,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct Authorization {
+    pub key: String,
+    pub value: String,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct Webhook {
+    pub url: String,
+    pub method: HttpMethod,
+    pub headers: Vec<HeadersEnum>,
+    pub authorization: Value,
+}
+
+#[derive(Serialize, Deserialize)]
+pub enum WebhookEvent {
+    ExperimentCreated,
+    ExperimentStarted,
+    ExperimentInprogress,
+    ExperimentUpdated,
+    ExperimentConcluded,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct WebhookResponse<T> {
+    pub event: WebhookEvent,
+    pub payload: T,
+}
+
+#[derive(Clone, Serialize, Deserialize)]
 pub struct TenantConfig {
     pub mandatory_dimensions: Vec<String>,
+    pub experiments_webhook_config: Value,
```

Comment:
we know the type of data here, so lets not use Value here and lets use a proper concrete type, which should be `Webhook` type, right ?
```suggestion
    pub webhook_config: Webhook,
```